warehouse/packaging/services.py

```diff
@@ -443,32 +447,34 @@ def _hit_ratelimits(self, request, creator):
         self.ratelimiters["project.create.user"].hit(creator.id)
         self.ratelimiters["project.create.ip"].hit(request.remote_addr)
 
-    def check_project_name(self, name: str) -> ProjectNameUnavailableReason | None:
+    def check_project_name(self, name: str) -> None:
         if not PROJECT_NAME_RE.match(name):
-            return ProjectNameUnavailableReason.Invalid
+            raise ProjectNameUnavailableInvalid()
 
         # Also check for collisions with Python Standard Library modules.
         if canonicalize_name(name) in STDLIB_PROHIBITED:
-            return ProjectNameUnavailableReason.Stdlib
+            raise ProjectNameUnavailableStdlib()
 
-        if self.db.query(
-            exists().where(Project.normalized_name == func.normalize_pep426_name(name))
-        ).scalar():
-            return ProjectNameUnavailableReason.AlreadyExists
+        if (
+            existing_project := self.db.query(Project)
+            .where(Project.normalized_name == func.normalize_pep426_name(name))
+            .first()
```

Comment:
suggestion: While we're here, use similar `db.scalars(select..` pattern that is used later for `similar_project_name`, so we have a consistent query interface. We should be moving to the `select()` style pattern everywhere eventually.

Reply:
Fixed